storage/local/persistence.go

```diff
@@ -114,10 +121,49 @@ type persistence struct {
 
 // newPersistence returns a newly allocated persistence backed by local disk storage, ready to use.
 func newPersistence(basePath string, chunkLen int, dirty bool) (*persistence, error) {
-	if err := os.MkdirAll(basePath, 0700); err != nil {
+	dirtyPath := filepath.Join(basePath, dirtyFileName)
+	versionPath := filepath.Join(basePath, versionFileName)
+
+	if file, err := os.Open(versionPath); err == nil {
+		defer file.Close()
+		data := make([]byte, 8)
+		n, err := file.Read(data)
```

Comment:
If you use http://golang.org/pkg/io/ioutil/#ReadFile you'll get the whole file returned guaranteed.